### PR TITLE
Strip excess custom exceptions from 'gcloud.streaming'

### LINKED ISSUE #1227: Strip excess custom exceptions from 'gcloud.streaming'
Many (all, maybe) of the custom exception types in `gcloud.streaming.exceptions` are providing no value, and could be replaced with either standard exceptions (`ValueError`, `TypeError`, etc.), or one of our existing types mapping particular HTTP response codes.

Strip out as many of the custom types as seems feasible, and move the remainder to `gcloud.exceptions`.

### PR DESCRIPTION
**Pull request by tseaver**
Closes #1227

### CODE REVIEW

File: gcloud/streaming/transfer.py [hunk 1]
```diff
@@ -647,11 +643,11 @@ def from_file(cls, filename, mime_type=None, auto_transfer=True, **kwds):
         """
         path = os.path.expanduser(filename)
         if not os.path.exists(path):
-            raise NotFoundError('Could not find file %s' % path)
+            raise ValueError('Could not find file %s' % path)
```
Comment (dhermes) [inline gcloud/streaming/transfer.py]: Shouldn't this be an `OSError`?

**Comment by dhermes** [comment]
This doesn't "move the remainder to `gcloud.exceptions`" as the bug says will happen.

File: gcloud/streaming/transfer.py (hunk 1 above)
Comment (tseaver) [inline gcloud/streaming/transfer.py]: I don't think so:  `OSError` [maps an actual `errno` returned from a system call](https://docs.python.org/2/library/exceptions.html#exceptions.OSError).

**Comment by tseaver** [comment]
The residue are all tangled up with fixing #1223.  If there are any left after that one, then they can move.

File: gcloud/streaming/transfer.py (hunk 1 above)
Comment (tseaver) [inline gcloud/streaming/transfer.py]: We could remove the "look before you leap" and just let `open(path, 'rb')` raise an `OSError`.

**Comment by dhermes** [comment]
I suggest filing another bug then (or leaving that one open and renaming it) so it doesn't slip our minds.

File: gcloud/streaming/transfer.py (hunk 1 above)
Comment (dhermes) [inline gcloud/streaming/transfer.py]: :+1:

**Comment by tseaver** [comment]
#1234 captures the final cleanup.

**Comment by dhermes** [comment]
Good deal

File: gcloud/streaming/transfer.py (hunk 1 above)
Comment (tseaver) [inline gcloud/streaming/transfer.py]: 7ddd0b0

**Comment by dhermes** [comment]
LGTM pending Travis

**Pull request merged**